packages/aic/src/db.ts

```diff
@@ -0,0 +1,69 @@
+import {knex, Knex} from 'knex'
+
+export type Database = {
+    /**
+     * The Database implementation must have two functions
+     * 
+     *  tickForDid to retreve the latest tick
+     *  putTickForDid to atomicly update the tick only if prevTid is the head
+     * 
+     */
+    tickForDid: ((did: string) => Promise<{ did: string; tid: string; tick: string } | undefined>),
+    putTickForDid: ((
+        did:string,
+        tid:string,  // the new tid must me larger then the previous tid
+        prevTid:string,  // gard: if the prevTid has changed the tick from the db is stale
+        tick:string  // the tick is a json string
+    ) => Promise<void>),
+}
+
+export const database = async (location: string): Promise<Database> =>  {
```

Comment:
Bigger tabs than elsewhere in the repo. Think prettier should fix this for you 🤔 
I can help get that set up if you like.

I also ran `yarn lint` & it popped up some linting errors that should be fixed (nothing major).

I may need to take a look again at how these tools interact w one another & if we can just go all in on one 🤔 

It's nice to have them setup with auto-formatting (similar to go) so you just don't ever have to think about it